larray/core/session.py

```diff
@@ -1124,21 +1131,26 @@ def summary(self, template=None):
         if LArray not in template:
             template[LArray] = "{key}: {axes_names} ({shape}) [{dtype}]\n    {title}"
 
-        def kind_kwargs(k, v):
-            if isinstance(v, Axis):
-                return (Axis, {'key': k, 'name': v.name, 'labels': v.labels_summary(), 'length': len(v)})
-            elif isinstance(v, Group):
-                return (Group, {'key': k, 'name': v.name, 'axis_name': v.axis.name, 'labels': v.key,
-                                  'length': len(v)})
-            elif isinstance(v, LArray):
-                return (LArray, {'key': k, 'axes_names': ', '.join(v.axes.display_names),
-                                  'shape': ' x '.join(str(i) for i in v.shape), 'title': v.title, 'dtype': v.dtype})
+        def display(k, v):
+            t = Group if isinstance(v, Group) else type(v)
+            tmpl = template.get(t, None)
+            if not (isinstance(tmpl, basestring) or callable(tmpl)):
```

Comment:
Wouldn't using a default template for unknown types: "{key}: {value}" be more useful?